src/Payum/Core/Bridge/Propel/Storage/PropelStorage.php

```diff
@@ -0,0 +1,89 @@
+<?php
+namespace Payum\Core\Bridge\Propel\Storage;
+
+use Payum\Core\Storage\AbstractStorage;
+use Payum\Core\Exception\LogicException;
+use Payum\Core\Model\Identity;
+
+class PropelStorage extends AbstractStorage
+{
+    /**
+     * @var string
+     */
+    protected $modelQuery;
+    /**
+     * @var string
+     */
+    protected $modelPeer;
+
+    /**
+     * @param string $modelClass
+     */
+    public function __construct($modelClass)
+    {
+        parent::__construct($modelClass);
+
+        $this->modelQuery = $modelClass.'Query';
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function doFind($id)
+    {
+        $modelQuery = $this->modelQuery;
+        return $modelQuery::create()
+            ->findPk($id);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function doUpdateModel($model)
+    {
+        $model->save();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function doDeleteModel($model)
+    {
+        $model->delete();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function doGetIdentity($model)
+    {
+        $id = $this->getModelId($model);
+
+        if (count($id) > 1) {
+            throw new LogicException('Storage not support composite primary ids');
+        }
+
+        return new Identity(array_shift($id), $model);
+    }
+
+    /**
+     * @param object $model
+     *
+     * @return mixed
+     */
+    protected function getModelId($model)
+    {
+        $id = array();
+        $modelPeer = get_class($model).'Peer';
```

Comment:
I'd prefer to move it to constructor, even better if you allow a developer to pass it as a second argument. query could be passed same,


They could optional and user do not provide them then you can build it in constructor.